Services/PackageService.cs

```diff
@@ -0,0 +1,36 @@
+using Repositories.Entities;
+using Repositories.Interfaces;
+
+namespace Services
+{
+    public class PackageService : IPackageService
+    {
+        private readonly IPackageRepository _packageRepository;
+        private readonly ICourierRepository _courierRepository;
+
+        public PackageService(IPackageRepository packageRepository, ICourierRepository courierRepository)
+        {
+            _packageRepository = packageRepository;
+            _courierRepository = courierRepository;
+        }
+
+        public async Task<List<Package>> GetUnassignedAsync()
+        {
+            var packages = await _packageRepository.GetAsync();
+            var couriers = await _courierRepository.GetAsync();
```

Comment:
I think it is not necessary to get that complicated. Why just not remove packages where courier is null?
And by the, we lack IQuerable (that of course, is not your part of job). Our data is in list, if we had iquarable it would be much simpler to of where statement and get only what we need. We will probably need to consider using IQuerable.

Reply:
Just checked we don't have that property (the one where you can view the relation between package and courier on package site). I am suggesting adding it. That would make the code much simpler and quicker.